src/test/java/com/github/packageurl/internal/StringUtilBenchmark.java

```diff
@@ -54,43 +54,45 @@
 @OutputTimeUnit(TimeUnit.MICROSECONDS)
 @State(Scope.Benchmark)
 public class StringUtilBenchmark {
+    private static final Random RANDOM = new Random();
 
     private static final int DATA_COUNT = 1000;
+
     private static final int DECODED_LENGTH = 256;
+
     private static final byte[] UNRESERVED =
             "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~".getBytes(StandardCharsets.US_ASCII);
 
     @Param({"0", "0.1", "0.5"})
     private double nonAsciiProb;
 
-    private String[] decodedData;
-    private String[] encodedData;
+    private String[] decodedData = {};
+
+    private String[] encodedData = {};
 
     @Setup
     public void setup() {
-        decodedData = createDecodedData();
-        encodedData = encodeData(decodedData);
+        createDecodedData();
+        createEncodedData();
```

Comment:
I am not a big fan of methods that need to be called in a very precise order to work. What about merging these methods?